plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/Utils.java

```diff
@@ -0,0 +1,63 @@
+package org.sonatype.nexus.repository.conda.internal.hosted;
+
+import groovy.lang.Tuple2;
+import org.apache.commons.lang.StringUtils;
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+import org.sonatype.nexus.repository.conda.internal.hosted.metadata.PackageDesc;
+import org.sonatype.nexus.repository.storage.Asset;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+
+public class Utils {
+
+    public static Tuple2<String, PackageDesc> toPackageDesc(Asset asset) {
```

Comment:
Question: Holy Tuple! Do you need to return a Tuple? It seems like you could get the filename using `CondaPath` where you are calling it? We tend to prefer plain ole Java for the most part, so adding in some Groovy here is a bit interesting.

Reply:
plain ole Java is really limiting. 
I wanted to return two objects out of this method and have a support from Type system. 

If you are not comfortable with this approach just let me know - I can change this some wrapper class or any other approach.

Reply:
You could use Pair from apache common if you want: https://commons.apache.org/proper/commons-lang/javadocs/api-3.9/org/apache/commons/lang3/tuple/Pair.html